tl

```diff
@@ -498,6 +621,23 @@ local function type_check_and_load(filename, modules)
    return chunk
 end
 
+--------------------------------------------------------------------
+--                        RUN BUILD SCRIPT                        --
+--------------------------------------------------------------------
+local script = {}
+if tlconfig.build_file then
+   script = type_check_and_load(tlconfig.build_file,modules)()
+end
+
+if script["gen_code"] then
+   --that the directory exposed to the script has the same name as the the folder it ends up at is kind of a hack (that I accidentally stumbled on)
+   --however, it is also a usefull hack as it provides a way to store the last build time without the 2 locations interfering.
+   --this is useful information so the result of the build.tl file can be cached, which can be benefical if the code generation takes a lot of time.
+   local full_path = path_concat(build_path, tlconfig["build_file_output_dir"] )
+   lfs.rmdir(full_path)
+   lfs.mkdir(full_path)
+   script["gen_code"](full_path)
```

Comment:
Again, should type check this manually even though we loaded a `.tl` file, we didn't inject any types into it. We have no idea if this is a function and should check for that. Then it should be `pcall`ed, since it is user code.